src/PackageProject.cmake

```diff
@@ -0,0 +1,143 @@
+# Uses ycm (permissive BSD-3-Clause license) and FowardArguments (permissive MIT license)
+
+macro(package_project targets)
+  set(_options ARCH_INDEPENDENT # default to false
+  )
+  set(_oneValueArgs
+      # default to the project_name:
+      NAME
+      COMPONENT
+      # default to project version:
+      VERSION
+      # default to any newer:
+      COMPATIBILITY
+      # default to ${CMAKE_BINARY_DIR}
+      CONFIG_EXPORT_DESTINATION
+      # default to ${CMAKE_INSTALL_DATADIR}/cmake/${NAME} suitable for vcpkg, etc.
+      CONFIG_INSTALL_DESTINATION)
+  set(_multiValueArgs
+      # a list of public/interface include directories or files
+      PUBLIC_INCLUDES
+      # the names of the INTERFACE/PUBLIC dependencies that are found using `CONFIG`
+      PUBLIC_DEPENDENCIES_CONFIGED
+      # the INTERFACE/PUBLIC dependencies that are found by any means using `find_dependency`.
+      # the arguments must be specified within double quotes (e.g. "<dependency> 1.0.0 EXACT" or "<dependency> CONFIG").
+      PUBLIC_DEPENDENCIES
+      # the names of the PRIVATE dependencies that are found using `CONFIG`. Only included when BUILD_SHARED_LIBS is OFF.
+      PRIVATE_DEPENDENCIES_CONFIGED
+      # PRIVATE dependencies that are only included when BUILD_SHARED_LIBS is OFF
+      PRIVATE_DEPENDENCIES)
```

Comment:
I wonder if these can be automatically found from the properties of the targets.